### PR TITLE
refactor(kit,nuxt,schema): use `consola` and improve test dx

### PR DESCRIPTION
**Pull request by pi0**
### 🔗 Linked issue

Extracted from #23225

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR refactors all packages to leverage consola logger exported from `@nuxt/kit`. 

We use a magical conosla wrapper in CLI that was automatically wrapping all console usages but it can be easily missed when running in tests or with programmatic usage.

Exceptions:
- For tests, we add a guard of `process.client` to avoid clattering output
- For schema package, we directly use consola to avoid kit dependency
- For nuxt/app and other runtime code, we use console for bundle size

Added back a (warning level) eslint rule also for `no-console`

**Note:** This PR also contains small other eslint tweaks to remove all current warnings.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
TLDR: Logs are still somehow leaking and spamming CI/Test output. Also, we have random failures with Ubuntu preset (most likely concurrency) but i gues it is worth to iterate over such big refactor into smaller steps...

File: test/utils.ts [hunk 1]
```diff
@@ -30,7 +30,11 @@ export async function renderPage (path = '/') {
     pageErrors.push(err)
   })
   page.on('request', (req) => {
-    requests.push(req.url().replace(url('/'), '/'))
+    try {
+      requests.push(req.url().replace(url('/'), '/'))
+    } catch (err) {
+      // TODO
```
Comment (danielroe) [inline test/utils.ts:36]: what kind of error was being triggered here?

Comment (pi0) [reply test/utils.ts:36]: The lovely context is not available one! Turns out we somehow still keep loading the Browser pages after tests are being done and we don't have context anymore.

This was more visible in new nuxi-edge case because we early close server properly now and made it more visible.

However i think would be nice later investigate root causes. Seems we are wasting too much on browser allocation in our tests.

**Review by danielroe — APPROVED** [review]

File: test/utils.ts (hunk 1 above)
Comment (danielroe) [reply test/utils.ts:36]: nice find!
